### ISSUE TITLE
Add support for single-threaded fan out and merge in

### ISSUE THREAD

**Issue by johnmcclean-aol** [aol/cyclops-react#472]
e.g.

```java
ReactiveSeq.of(1,2,3,4)
           .fanOut(s1->s1.filter(i->i%2==0).map(i->i*2),
                   s2->s2.filter(i->i%2!=0).map(i->i*100))
           .toListX();
```
Should result in 

```java
ListX.of(4,100,8,300))
```

Should work for all of

1. Push based reactive Spouts
2. Pull based synchronous Streams
3. FutureStreams

**Comment by johnmcclean-aol** [comment]
Add similar issues for parallelFanOut and fanOutZipIn - both sequential and parallel

**Comment by johnmcclean-aol** [comment]
Across all 3 Stream types.

**Comment by johnmcclean-aol** [comment]
Released in 2.0.0-MI3
